src/main/java/frc/robot/util/AutoLogic.java

```diff
@@ -0,0 +1,235 @@
+
+      DriverStation.reportError("Ooofs: " + e.getMessage(), e.getStackTrace());
+    }
+
+    // Create a path following command using AutoBuilder. This will also trigger event markers.
+
+    return null;
+  }
+
+  public static void initShuffleBoard() {
+
+    addAutoOptions();
+    addTestedAutos();
+    tab.add("Auto Selector", autoPicker)
+        .withWidget(BuiltInWidgets.kComboBoxChooser)
+        .withPosition(0, 0)
+        .withSize(2, 1);
+
+    tab.add("Elevator Commands", autoAction)
+        .withWidget(BuiltInWidgets.kComboBoxChooser)
+        .withPosition(3, 2)
+        .withSize(2, 1);
+
+    tab.addString("Current Selected path", () -> getSelectedAutoName())
+        .withPosition(0, 1)
+        .withSize(2, 1);
+
+    tab.add("Tested Autos", testedAutos)
+        .withWidget(BuiltInWidgets.kComboBoxChooser)
+        .withPosition(3, 0)
+        .withSize(2, 1);
+
+    autoAction.setDefaultOption("EXTEND WHILE MOVING", "raiseElevator");
+    autoAction.addOption("EXTEND AND WAIT", "raiseElevatorandWait");
+  }
+
+  public static <T> String[] toStringArray(List<T> dataList) {
+
+    String[] data = new String[dataList.size()];
+
+    for (int i = 0; i < dataList.size(); i++) {
+
+      String addedData = dataList.get(i).toString();
+
+      data[i] = "\n" + addedData;
+    }
+
+    return data;
+  }
+
+  public static PathPlannerTrajectory makeTrajectory(
+      PathPlannerPath path,
+      ChassisSpeeds startingSpeeds,
+      Rotation2d startingRotation,
+      RobotConfig config) {
+
+    PathPlannerTrajectory trajectory =
+        new PathPlannerTrajectory(path, startingSpeeds, startingRotation, config);
+
+    return trajectory;
+  }
+
+  public static void addAutoOptions() {
+    autoPicker.setDefaultOption("DEFAULT PATH", "PreLoadAuto");
+    autoPicker.addOption("LOW TWO PIECE", "OppsidetoDandC");
+    autoPicker.addOption("TEST LOW TWO PIECE", "Copy of OppsidetoDandC");
+    autoPicker.addOption("INTAKE TEST", "INTAKE TEST");
+    autoPicker.addOption("Mid 3 Piece", "Mid3Piece");
+    autoPicker.addOption("Low 3 Piece", "Low3Piece");
+    autoPicker.addOption("DRIVE ONLY", "SideWall");
+    autoPicker.addOption("Low  1 L4", "OppsidetoD");
+    autoPicker.addOption("Upper 3 Piece", "Upper3Piece");
+    autoPicker.addOption("CHOREO CRAZY TEST?(IGNORE)", "4 L4 Coral");
+    autoPicker.addOption("CHOREO 3 PIECE?(IGNORE)", "New Choreo");
+    autoPicker.addOption("CHOREO 3 PIECE LESS CRAZY?(IGNORE)", "Triple7");
+    autoPicker.addOption("High 1 L4", "SideToJ");
+    autoPicker.addOption("CHOREO High 4 L4(IGNORE)", "HighL4");
+    autoPicker.addOption("TESTING Low 3 Piece", "TEST Low3Piece");
+    autoPicker.addOption("ALIGNMENT TEST", "ALIGN TEST");
+    autoPicker.addOption("DRIVE TEST", "DRIVE1FT");
+  }
+
+  public static void addTestedAutos() {}
+
+  public static Command raiseElevator() {
+    if (r.superStructure != null) {
+      return r.superStructure.levelFour(() -> true).asProxy();
```

Comment:
`.asProxy()` should only be needed if this is inside a composite command (parallel, sequential, etc.). Since that's unknowable here, I'd recommend removing it here & adding it in whatever composite command is calling this (and only if there are no other options to avoid its use).